doc/cmis-module-enhancement/cmis-module-enhancement.md

```diff
@@ -0,0 +1,224 @@
+# Enhancement of CMIS module management
+
+## Table of Content 
+
+## 1. Revision 
+
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 | July 2023   |  Noa Or            | Initial version                   |
+
+## 2. Scope  
+
+This section describes an enhancment of the synchronization between ASIC port and module configuration.
+
+## 3. Definitions/Abbreviations 
+
+N/A
+
+## 4. Overview
+
+Configuration of the ASIC side (port SERDES) is handled by the SWSS docker (Ports Orch Agent) interacting with the vendor SAI via SAI calls, whereas the modules supporting the CMIS protocol are configured by PMON docker (Xcvrd daemon).
+
+These initialization processes should be synchronized and the configuration of CMIS module should start only after the ASIC is initialized and per CMIS 5.2 it started sending the high-speed signal toward a module.
+
+Currently, SONIC uses the "host_tx_ready"  flag in the PORT table in STATE DB for synchronization. This flag is set by Ports OA right after the SAI API for setting the Admin status to UP returns with OK/Success status. PMON registers for changes of this flag in Redis DB and starts the CMIS initialization for a particular module when this flag is set.
+
+Current design has some gaps with synchronization between ASIC and module configuration.
+This document purpose is to introduce an enhacement that will address the gaps and find a backward compatible solution. 
+
+
+## 5. Requirements
+
+* SONiC shall have backward compatibility for platforms that don't support the proposed enhancement.
+
+* Vendor SDK/FW shall start sending high-speed signal to module, only when admin status is up, and it has an indication that module was plugged in.
+
+* Vendor SDK/FW shall support asynchronous notification of start/stop of sending high-speed signal from ASIC to module.
+
+* PortsOrch shall set host_tx_ready in state DB, only when it recieved notification that the high-speed signal is sent. 
+
+## 6. High-Level Design 
+
+Current Flow:
+<p align=left>
+<img src="img/The-Current-Flow.png" alt="Figure 1. Current Flow">
+</p>
+
+### 6.1. Problem Statement
+With move to SW-based management of CMIS modules - when the ASIC side is configured by vendor SDK, and module side by SONIC, some problems have been identified with the current approach.
+
+#### 6.1.2. Host TX Ready signal
+As mentioned earlier, currently SONIC assumes that as soon as the Admin Status is set to UP and the corresponding SAI call returns with OK/SUCCESS status, the "host_tx_ready" flag can be set in the STATE DB to trigger the CMIS State Machine for specific port. 
+But it is not always a truth. ASIC port initialization process takes some time and this time can increase with move to new transceiver technologies (e.g. BiDi). 
```

Comment:
SONiC taken (~ 10-20 sec )to bring up the ports and complete the FSM, do you see any performance impact with this design tx_ready_signal ? 